lib/mcts/node.rb

```diff
@@ -40,43 +40,32 @@ def expand
     end
 
     def rollout
-      playout = Playout.new(@game_state)
-      playout.play
+      state = @game_state.dup
+      state.set_move(state.generate_move) until state.finished?
+      state.winner
     end
 
-    def won
-      @visits += 1
-      @wins += 1
-    end
-
-    def lost
-      @visits += 1
-    end
-
-    def backpropagate(won)
+    def backpropagate(winner:)
       node = self
-      node.update_won won
-      until node.root? do
-        won = !won # switching players perspective
-        node = node.parent
-        node.update_won(won)
-      end
+      node.update_stats(winner: winner) until (node = node.parent).nil?
     end
 
     def untried_moves?
       !@untried_moves.empty?
     end
 
-    def update_won(won)
-      if won
-        self.won
-      else
-        self.lost
-      end
+    def update_stats(winner:)
+      @visits += 1
+      @wins += 1 if winner == acting_player
     end
 
     private
 
+    def acting_player
+      return -1 if parent.nil?
```

Comment:
Why do we return `-1` here under these circumstances? Shouldn't we always be able to just return the last_turn_color? Plus, if we wanna exit early I believe `false/nil` are better values :)

Reply:
Oh, i just wanted it to always return a number which would never == winner. I ran into an issue in a multi-player game where on the first move of the game there wasn't a previous last_turn_color.

I'm cool with returning `nil`; just a personal habit to avoid returning nil in languages without optionals :)